Bottom: 4900b7f1c234aeea834da48be81ce9aa7d5eb7cf
Top:    d2d2277ee9ee3f7b2c7044b591fb5da075684f1a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-26 16:13:14 +0000

Remove hack in draw.c which broke PS (and probably other) output

The set_layer call hacked into DrawLayerGroup to reset compositing tricked
the PS exporter to re-open and overwrite output in the main layer output
file during the fab drawing output.


---

diff --git a/src/draw.c b/src/draw.c
index f7c1556..3c4d10f 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -881,8 +881,7 @@ DrawLayerGroup (int group, const BoxType * screen)
   Cardinal *layers = PCB->LayerGroups.Entries[group];
 
   clip_box = screen;
-  for (i = n_entries - 1; i >= 0;
-      i--, gui->set_layer (0, group, 0)) /* HACK: Subcomposite each layer in a layer group separately */
+  for (i = n_entries - 1; i >= 0; i--)
     {
       layernum = layers[i];
       Layer = PCB->Data->Layer + layers[i];
@@ -899,10 +898,6 @@ DrawLayerGroup (int group, const BoxType * screen)
 	      r_search (Layer->polygon_tree, screen, NULL, poly_callback,
 			&info);
 	      info.arg = false;
-
-	      /* HACK: Subcomposite polygons separately from other layer primitives */
-	      /* Reset the compositing */
-	      gui->set_layer (0, group, 0);
 	    }
 
 	  if (TEST_FLAG (CHECKPLANESFLAG, PCB))
